### ISSUE TITLE
[feature request] web3.ens.resolve should take a block number

### ISSUE THREAD

**Issue by WyseNynja** [ethereum/web3.py#1984]
* Version: 5.18.0
* Python: 3.9
* OS: osx

### What was wrong?

I am doing some analysis of some of my old transactions and noticed that `web3.ens.resolve` was returning the current address and not the address at the time of the transaction.

### How can it be fixed?

I think adding a kwarg to `web3.ens.resolve` that gets passed to the `call` functions inside would be a simple addition that would be backwards compatible.

**Comment by kclowes** [comment]
Good idea, thanks for the request! The fastest route to get this in would be to open up a PR that we can review.
